exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -354,12 +365,16 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 		})
 	}
 	if len(errList) > 0 {
-		return kerrors.NewAggregate(errList)
+		return ctrl.Result{}, kerrors.NewAggregate(errList)
 	}
 
 	conditions.MarkTrue(clusterResourceSet, addonsv1.ResourcesAppliedCondition)
 
-	return nil
+	if resourceNotFound {
```

Comment:
Q: should this go before setting the condition to true?

Reply:
As discussed move this up a bit